### ISSUE TITLE
[rl] C51: drop unnecessary probs_all.clone() in learn_step

### ISSUE THREAD

**Issue by anthonytorlucci** [anthonytorlucci/rlevo#419]
- **Files:** `algorithms/c51/c51_agent.rs`
- **Body:** `learn_step` computes `probs_all` at `c51_agent.rs:554` (`activation::softmax(logits.clone(), 2)`) and consumes it exactly once, at `:557`, via `probs_all.clone() * support_3d_auto`. Confirmed by grep: `probs_all` has no other use site in the file, so the `.clone()` at `:557` clones a `(B, A, N)` tensor for no reason — `probs_all` can be moved instead. (`logits.clone()` at `:554` is separately required, since `logits` is consumed by value later via `gather` at `:567` — leave that one.)
- **Suggested fix:**
  ```rust
  let q_all: Tensor<B, 2> = (probs_all * support_3d_auto).sum_dim(2).squeeze_dim::<2>(2);
  ```
- **Severity:** No High; a single unnecessary tensor clone in a hot path (every learn step). Low effort, low risk.
- **Source:** C51 code review (2026-07) §2.3.

**Comment by anthonytorlucci** [comment]
Related clone-reduction observation in the same \`learn_step\`, adjacent to this issue's \`probs_all\` clone: \`pred_log_p\` is cloned for the entropy diagnostic (\`c51_agent.rs:573-574\`, via \`.exp()\` and again into the entropy sum) before being cloned once more at the cross-entropy call site (\`:588\`) and moved into the KL call (\`:615\`). Binding \`let log_p = &pred_log_p;\` for the entropy block and moving the owned value into the loss/KL calls afterward would drop one of the three clones. Same low-severity, low-effort bucket as the \`probs_all\` fix above — noting it here rather than opening a second issue for the same hot-path-clone pattern.

Source: C51 code review (2026-07) §2.1.
